File: esw-integration-test/src/test/scala/esw/gateway/server/testdata/AssemblyBehaviourFactory.scala
package esw.gateway.server.testdata

import akka.actor.typed.scaladsl.ActorContext
import csw.command.client.messages.TopLevelActorMessage
import csw.framework.models.CswContext
import csw.framework.scaladsl.{ComponentBehaviorFactory, ComponentHandlers}

class AssemblyBehaviourFactory extends ComponentBehaviorFactory {
  override protected def handlers(ctx: ActorContext[TopLevelActorMessage], cswCtx: CswContext): ComponentHandlers =
    new SampleComponentHandlers(ctx, cswCtx)
}

File: esw-kt/integration-test/src/test/java/esw/ocs/testdata/AssemblyBehaviourFactory.java
package esw.ocs.testdata;

import csw.command.client.messages.TopLevelActorMessage;
import csw.framework.javadsl.JComponentBehaviorFactory;
import csw.framework.javadsl.JComponentHandlers;
import csw.framework.models.JCswContext;

public class AssemblyBehaviourFactory extends JComponentBehaviorFactory {
    @Override
    public JComponentHandlers jHandlers(akka.actor.typed.javadsl.ActorContext<TopLevelActorMessage> ctx, JCswContext cswCtx) {
        return new SampleComponentHandlers(ctx, cswCtx);
    }
}
